.github/workflows/test.yml

```diff
@@ -3,11 +3,55 @@ name: Tests
 on:
   pull_request:
   push:
-    branches: [master, develop]
+    branches: [develop]
 
 jobs:
-  test-semgrep:
-    name: Live runs of action
+  pytest:
+    name: run pytest
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [3.7, 3.8, 3.9]
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Setup Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v1
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - id: setup-poetry
+        name: Setup poetry
+        run: |
+          python -m pip install --upgrade poetry
+          echo "::set-output name=cache-dir::$(poetry config cache-dir)"
+
+      # if we cached the complete virtualenv, it would cache the local semgrep-agent package too
+      # so instead we cache poetry's package cache, which speeds up the install command
+      - id: reuse-poetry-package-cache
+        name: Reuse poetry's package cache
+        uses: actions/cache@v2
+        with:
+          path: ${{ steps.setup-poetry.outputs.cache-dir }}
+          key: ${{ runner.os }}-${{ matrix.python-version }}-poetry-${{ hashFiles('**/poetry.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-${{ matrix.python-version }}-poetry-
+
+      - name: Install dependencies
+        run: |
+          poetry install
+          poetry add semgrep --no-interaction
```

Comment:
1. the unpinned version bothers me
2. not sure if installing with poetry or with pipx (or pip) is better

Reply:
Why not just pin it? We update semgrep version manually anyway. Just add the it should also be bumped here?

Reply:
I think you'd want to update the bump script in that case, as well

Reply:
No bump script here unfortunately, and wanted to avoid adding more manual work :(